packages/@o3r/extractors/src/core/comparator/metadata-comparison.helper.ts

```diff
@@ -0,0 +1,116 @@
+import type { BuilderContext, BuilderOutput } from '@angular-devkit/architect';
+import { getPackageManagerInfo, type PackageManagerOptions, type SupportedPackageManagers, type WorkspaceSchema } from '@o3r/schematics';
+import { existsSync, readFileSync } from 'node:fs';
+import { join } from 'node:path';
+import type { MetadataComparator, MigrationData, MigrationMetadataCheckBuilderOptions } from './metadata-comparator.interface';
+import { getFilesFromRegistry, getLatestMigrationMetadataFile, getLocalMetadataFile, getVersionFromFilename, getVersionRangeFromLatestVersion } from './metadata-files.helper';
+
+// TODO : create a bespoke type for the errors?
+function checkMetadataFile(lastMetadataFile: any, newMetadataFile: any, migrationData: MigrationData<any>[], isBreakingChangeAllowed: boolean, comparator: MetadataComparator<any>): Error[] {
+  const errors = [] as Error[];
+  const newMetadataArray = comparator.getArray(newMetadataFile);
+  const lastMetadataArray = comparator.getArray(lastMetadataFile);
+  for (const lastValue of lastMetadataArray) {
+    const isInNewMetadata = newMetadataArray.some((newValue) => comparator.isSame(newValue, lastValue));
+    if (isInNewMetadata) {
+      if (!isBreakingChangeAllowed) {
+        errors.push(new Error(`Property ${comparator.getName(lastValue)} is not present in the new metadata and breaking changes are not allowed`));
+        break;
+      }
+
+      const migrationMetadataValue = migrationData.find((metadata) => comparator.isMigrationDataMatch(lastValue, metadata.after));
+
+      if (!migrationMetadataValue) {
+        errors.push(new Error(`Property ${comparator.getName(lastValue)} has been modified but is not documented in the migration document`));
+        break;
+      }
+
+      if (migrationMetadataValue.after) {
+        const isNewValueInNewMetadata = newMetadataArray.some((newValue) => comparator.isMigrationDataMatch(newValue, migrationMetadataValue.after));
+        if (!isNewValueInNewMetadata) {
+          errors.push(new Error(`Property ${comparator.getName(lastValue)} has been modified but the new property is not present in the new metadata`));
+          break;
+        }
+      }
+    }
+  }
+  return errors;
+}
+
+/**
+ * Gets the package manager to use to retrieve the previous package from npm.
+ * If the project uses npm or yarn it will be npm.
+ * If the project uses yarn 2+ it will be yarn.
+ * This is especially important because npm and yarn 1 use the authentication from the .npmrc while yarn 2+ uses the .yarnrc.
+ * @param options Option to determine the final package manager
+ */
+function getPackageManagerForRegistry(options?: PackageManagerOptions): SupportedPackageManagers | undefined {
+  const packageManagerInfo = getPackageManagerInfo(options);
+  if (!packageManagerInfo.version) {
```

Comment:
If it is `npm` without version, do we want to return `undefined`?

Reply:
can it happen ?